Mono.Cecil.Cil/MethodBody.cs

```diff
@@ -106,12 +108,25 @@ static ParameterDefinition CreateThisParameter (MethodDefinition method)
 		public MethodBody (MethodDefinition method)
 		{
 			this.method = method;
+			instructionTokens = new Dictionary<Instruction, MetadataToken> ();
 		}
 
 		public ILProcessor GetILProcessor ()
 		{
 			return new ILProcessor (this);
 		}
+
+		public bool GetInstructionToken (Instruction instruction, out MetadataToken token)
+		{
+			if (!instructionTokens.ContainsKey (instruction))
```

Comment:
I think you could use 
`return instructionTokens .TryGetValue(instruction, token);`